src/query-builder/SelectQueryBuilder.ts

```diff
@@ -2906,6 +2971,20 @@ export class SelectQueryBuilder<Entity>
         return parseInt(results[0]["cnt"])
     }
 
+    protected async executeExistsQuery(
+        queryRunner: QueryRunner,
+    ): Promise<boolean> {
+        const results = await this.connection
+            .createQueryBuilder()
+            .from("(SELECT 1 AS row_exists)", "row_exists_tbl")
```

Comment:
`FROM` clause is missing. Some dialects allow it, some not. Best thing is to add it manually:
```ts
qb
  .select("1", "row_exists")
  .from("SYS.DUMMY", "dummy_tbl")
  .whereExists(this)
```

Reply:
IMO you could use the query builder without a `from`: `qb.select("1", "row_exists").whereExists(this)`.

You would then need to update the return value of `SelectQueryBuilder#createSelectExpression()` like this:

- if `froms` is not empty, keep the current behavior (add the concatenated list). 
- if it is empty and driver is oracle or sap, then add `FROM DUAL` / `FROM SYS.DUMMY` (like in the `InsertQueryBuilder`).
- if it is empty and the driver is something else then don't add `FROM` at all.

Reply:
@alumni 

Thanks for the comment.

Initially, I tried your solution to handle the `exists` method. But it becomes very complex. I had to generate different queries for each driver (every driver had to implement a method for generating the right query), and the result value from each driver was different. in the end, It had lots of "if" statements.

I checked most of the drivers that Typeorm supports and came up with this query that is independent of the driver and the result from the drivers is the same for all of them.

Reply:
@mortezaPRK well, as I said, this is driver-specific, so you need to handle some drivers in a different way. This is nothing special, it's done in quite a lot of places in the project.

`(SELECT 1 as row_exists)` throws a `QueryFailedError` on an Oracle / SAP connection because the `FROM` clause is missing, so special handling for them would be needed. And I would do this special handling in `SelectQueryBuilder#createSelectExpression()`, it would also allow selecting from dummy/dual inside the apps that use TypeORM.

Maybe one of the maintainers has a better idea here :)

Reply:
Agree :)

I'll change the implementation ASAP

Reply:
@alumni

I added `fromDummy` and it works perfectly find :)

Thanks for the idea.